lib/amqp.js

```diff
@@ -138,16 +138,37 @@ class Amqp {
         log.trace('Pushing to exchange=%s, routingKey=%s, messageSize=%d, options=%j',
             exchangeName, routingKey, payload.length, options);
         log.debug('Current memory usage: %s Mb', process.memoryUsage().heapUsed / 1048576);
-
         return this.publishMessage(exchangeName, routingKey, buffer, options, 0);
     }
 
     async publishMessage(exchangeName, routingKey, payloadBuffer, options, iteration) {
         const settings = this.settings;
-        const publishChannel = this.publishChannel;
+        try {
+            const result = await this._promisifiedPublish(exchangeName, routingKey, payloadBuffer, options);
+            if (!result) {
+                log.warn('Buffer full when publishing a message to '
+                    + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
+            }
+            return result;
+        } catch (error) {
+            log.error(error, 'Failed on publishing message to queue');
+            const currentDelay = settings.AMQP_PUBLISH_RETRY_DELAY;
+            const maxDelay = settings.AMQP_PUBLISH_MAX_RETRY_DELAY;
+            const delay = Math.min(currentDelay * iteration, maxDelay);
+            await new Promise(resolve => setTimeout(resolve, delay));
```

Comment:
please, add some `trace` log message with `iteration`, `delay`, .... for easy debugging.